File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
        unordered_map<int,int> table;
        
        int sum = 0, res = 0;
        table[0] = 1;
 
        for(int i = 0; i < nums.size(); i++){
            
            sum = sum + nums[i];
            
            if(table.find(sum - k) != table.end())
                res += table[sum-k];
            
            table[sum]++;
            
        }
        
        return res;
        
    }
};

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        
        total = 0
        res = 0
        table = {}
        table[0] = 1
        
        for items in nums:
            
            total += items
            
            if table.get(total) == None:
    
                table[total] = 0
            
            if table.get(total-k) != None:
                
                res += table[total-k]

            table[total] += 1
            
        return res